File: src/main/java/ua/notion/infrastructure/persistence/contract/WeatherEventRepository.java
package ua.notion.infrastructure.persistence.contract;

import java.util.List;
import ua.notion.domain.entity.WeatherEvent;
import ua.notion.infrastructure.persistence.Repository;

public interface WeatherEventRepository extends Repository<WeatherEvent, Integer> {

  List<WeatherEvent> findByGardenId(Integer gardenId);

  List<WeatherEvent> findByEventType(String eventType);

  long countByGardenId(Integer gardenId);
}

File: src/server/src/repositories/contract/weather_event_repository.rs
use axum::async_trait;
use chrono::{DateTime, Utc};

use crate::{
    errors::error::DbResult,
    models::{EventType, WeatherEvent},
    repositories::repository::Repository,
};

#[async_trait]
pub trait WeatherEventRepository: Repository<WeatherEvent, i32> {
    async fn find_by_garden_id(&self, garden_id: i32) -> DbResult<Vec<WeatherEvent>>;
    async fn find_by_event_type(&self, event_type: EventType) -> DbResult<Vec<WeatherEvent>>;
    async fn find_by_garden_id_and_event_type(
        &self,
        garden_id: i32,
        event_type: EventType,
    ) -> DbResult<Vec<WeatherEvent>>;
    async fn find_by_garden_id_since(
        &self,
        garden_id: i32,
        since: DateTime<Utc>,
    ) -> DbResult<Vec<WeatherEvent>>;
}
